src/Model/HasLegacyAttributesTrait.php

```diff
@@ -64,6 +76,13 @@ public function getAttribute(string $key)
     {
         $inAttributes = array_key_exists($key, $this->attributes);
 
+        if ($caster = CastResolver::resolve($this->casts[$key] ?? null)) {
+            $value = $caster->get($this->attributes[$key] ?? null);
+            $this->attributes[$key] = $value;
```

Comment:
This works same way of the `setAttribute` method.
Is that right? Makes sense do it this here?

Reply:
It works almost in the same way as `setAttribute`. 
The difference is that here we are using `get` method from Caster. There we are using `set`.


In Legacy Mongolid Attributes, we must overwrite the `$this->attributes` property in order to add this custom cast behaviour. Then these methods (`getAttibute` and `setAttribute`) will be used internally by Mongolid when retrieving or persisting the entity.

Reply:
esse get não parece estar certo.

Deveria apenas retornar o valor.